### DISCUSSION TITLE
Transaction encoding to String for sending over

### DISCUSSION THREAD

**Discussion by degenSumi** [aptos-labs/aptos-developer-discussions#167] [Questions] [answered]
How do i convert this transaction to a string so that i can send it over an api and then convert it back to a raw transaction to sign it ?
can you please help me, i went through the sdk but could'nt find any function that do this?
const txn = await aptos.transaction.build.simple({
    sender: "0xbe5cd08d8fe346b9b55dba64fc09e50dda92a8bd16743652ec8b156821123310",
    data: {
      function: "0xc7efb4076dbe143cbcd98cfaaa929ecfc8f299203dfff63b95ccb6bfe19850fa::router::swap_exact_input",
      typeArguments: [parseTypeTag(APTOS_COIN), parseTypeTag(DEVNET_BNB)],
      functionArguments: [new U64(APTOS_COIN_AMOUNT), new U64(5)],
    },
  });

  console.log(txn);

// Something like 
const serialisedTx = tx.toString("base64")
const txRaw = Transaction.from(tx)

**Comment by gregnazario** [comment, marked as answer]
Here's a good example of how to serialize the transaction for sending to an external signer.  We convert it to BCS bytes and then convert it back

https://github.com/aptos-labs/aptos-ts-sdk/blob/main/examples/typescript/external_signing.ts#L152-L157

**Reply by Hjghvggjn** [reply]
Pls I tried sending fund  from My aptos wallet in trust wallet to bitget wallet to another aptos wallet in my bitget wallet but the transaction failed till now I didn't receive my fund back in my aptos wallet in trust wallet,pls how do I have my fund back in my trust wallet aptos wallet
